File: src/go/efitoggler/efitoggler.go
package main

import (
	"efitoggler/lib"
	"fmt"
	"os"
	"os/exec"
	"runtime"
	"strings"
	"time"

	"github.com/AlecAivazis/survey/v2"
	"github.com/briandowns/spinner"
)

func core() {
	lib.Clear(); language(); lib.Cover(); verify(); toggler()
}

var LANGUAGE = 0

func language() {
	option := 0
	promptLang := &survey.Select{
		Message: "Bienvenido / Welcome \n Choose your language / Selecciona tu idioma",
		Options: []string{"English", "Espanol"},
	}
	survey.AskOne(promptLang, &option)

	if option == 0 {
		LANGUAGE = 1
	} else {
		LANGUAGE = 2
	}
}

func verify() {
	platform := runtime.GOOS
	if platform != "darwin" {
		lib.Clear()
		lib.Printer("error", 0, LANGUAGE)
		fmt.Print("\n")
		os.Exit(1)
	}
	lib.Printer("print", 1, LANGUAGE)
	s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
	s.Start(); time.Sleep(time.Second); s.Stop()
}

func toggler() {

	sout1, err1 := exec.Command("bash", "-c",
		"diskutil list | sed -ne '/EFI/p' | sed -ne 's/.*\\(d.*\\).*/\\1/p'").Output()
	if err1 != nil {fmt.Println(err1); os.Exit(1)}

	sout2, err2 := exec.Command("bash", "-c",
		"EFIPART=$(diskutil list | sed -ne '/EFI/p' | sed -ne 's/.*\\(d.*\\).*/\\1/p') MOUNTROOT=$(df -h | sed -ne \"/$EFIPART/p\"); echo $MOUNTROOT").Output()
	if err2 != nil {fmt.Println(err2); os.Exit(1)}

	EFIPART, EFI := string(sout1), string(sout2)

	EFIPART = strings.TrimSuffix(EFIPART, "\n")
	EFI = strings.TrimSuffix(EFI, "\n")

	if EFI != "" {
		lib.Printer("print", 2, LANGUAGE)
		checkDev := exec.Command("bash", "-c", "sudo cat < /dev/null")
		checkDev.Stdin = os.Stdin
		outChck, errChck := checkDev.Output()
		if errChck == nil {
			_ = outChck
			exec.Command("bash", "-c", fmt.Sprintf("sudo diskutil unmount %s",EFIPART)).Run()
			exec.Command("bash", "-c", "sudo rm -rf /Volumes/EFI").Run()
			lib.Clear()
			lib.Printer("print", 4, LANGUAGE)
		} else {
			lib.Clear()
			lib.Printer("print", 6, LANGUAGE)
			fmt.Println("")
			os.Exit(1)
		}
	} else {
		lib.Printer("print", 3, LANGUAGE)
		checkDev := exec.Command("bash", "-c", "sudo cat < /dev/null")
		checkDev.Stdin = os.Stdin
		outChck, errChck := checkDev.Output()
		if errChck == nil {
			_ = outChck
			exec.Command("bash", "-c", "sudo mkdir /Volumes/EFI").Run()
			exec.Command("bash", "-c", fmt.Sprintf("sudo mount -t msdos /dev/%s /Volumes/EFI", EFIPART)).Run()
			exec.Command("bash", "-c", "open /Volumes/EFI").Run()
			lib.Clear()
			lib.Printer("print", 4, LANGUAGE)
		} else {
			lib.Clear()
			lib.Printer("print", 6, LANGUAGE)
			fmt.Println("")
			os.Exit(1)
		}
	}
}

func main() { core() }

File: src/dart/efitoggler/bin/efitoggler.dart
import 'dart:io' show Platform, exit;

import 'package:process_run/shell_run.dart';

import 'package:interact/interact.dart';

import 'package:efitoggler/index.dart';

void core() {
  clear(); language(); cover(); verify();
}

int _language = 0;

void language() {
  print("Bienvenido / Welcome");
  final languages = ['English', 'Espanol'];
  final selection = Select(
    prompt: 'Please, choose your language / Por favor selecciona tu idioma',
    options: languages
  ).interact();

  selection == 0 ? _language = 1 : _language = 2;
}

void verify() {
  if(!Platform.isMacOS) {
    clear(); printer("error", 0, _language);
    print("\n"); exit(1);
  }
  printer("print", 1, _language);
  spin(() => toggler());
}

void toggler() async {

  final shell = Shell(throwOnError: false, verbose: false);

  final result1 = await shell.runExecutableArguments("bash", [
    "-c", "diskutil list | sed -ne '/EFI/p' | sed -ne 's/.*\\(d.*\\).*/\\1/p'"
  ]);

  if(result1.exitCode != 0) exit(1);

  final result2 = await shell.runExecutableArguments("bash",[
    "-c","EFIPART=\$(diskutil list | sed -ne '/EFI/p' | sed -ne 's/.*\\(d.*\\).*/\\1/p') MOUNTROOT=\$(df -h | sed -ne \"/\$EFIPART/p\"); echo \$MOUNTROOT"
  ]);

  if(result2.exitCode != 0) exit(1);

  final efipart = (result1.stdout as String).trim();
  final efi = (result2.stdout as String).trim();

  if(efi != "") {
    printer("print", 2, _language);

  final checkDev = await shell.runExecutableArguments("bash", ["-c", "sudo cat < /dev/null"]);
    if(checkDev.exitCode == 0) {
      await shell.runExecutableArguments("bash", ["-c", "sudo diskutil unmount $efipart"]);
      await shell.runExecutableArguments("bash", ["-c", "sudo rm -rf /Volumes/EFI"]);
      clear(); printer("print", 4, _language);
    } else {
      clear(); printer("error", 5, _language); exit(1);
    }
  } else {
    printer("print", 3, _language);
    final checkDev = await shell.runExecutableArguments("bash", ["-c", "sudo cat < /dev/null"]);
    if(checkDev.exitCode == 0) {
      await shell.runExecutableArguments("bash", ["-c", "sudo mkdir /Volumes/EFI"]);
      await shell.runExecutableArguments("bash", ["-c", "sudo mount -t msdos /dev/$efipart /Volumes/EFI"]);
      await shell.runExecutableArguments("bash", ["-c", "open /Volumes/EFI"]);
      clear(); printer("print", 4, _language);
    } else {
      clear(); printer("error", 5, _language); exit(1);
    }
  }
}

void main() { core(); }